engine/executor/dm/unitholder.go

```diff
@@ -243,22 +248,28 @@ func (u *unitHolderImpl) Status(ctx context.Context) interface{} {
 	if err != nil {
 		u.logger.Warn("failed to get source status", zap.Error(err))
 	}
-	u.sourceStatusMu.Lock()
-	u.sourceStatus = sourceStatus
-	u.sourceStatusMu.Unlock()
+	u.setSourceStatus(sourceStatus)
+	return nil
+}
 
-	// nil sourceStatus is supported
-	return u.unit.Status(sourceStatus)
+func (u *unitHolderImpl) getSourceStatus() *binlog.SourceStatus {
+	u.sourceStatusMu.RLock()
+	defer u.sourceStatusMu.RUnlock()
+	return u.sourceStatus
+}
+
+func (u *unitHolderImpl) setSourceStatus(in *binlog.SourceStatus) {
+	u.sourceStatusMu.Lock()
+	defer u.sourceStatusMu.Unlock()
+	u.sourceStatus = in
 }
 
 // CheckAndUpdateStatus implement UnitHolder.CheckAndUpdateStatus.
 func (u *unitHolderImpl) CheckAndUpdateStatus(ctx context.Context) {
-	u.sourceStatusMu.RLock()
-	sourceStatus := u.sourceStatus
-	u.sourceStatusMu.RUnlock()
+	sourceStatus := u.getSourceStatus()
 
 	if sourceStatus == nil || time.Since(sourceStatus.UpdateTime) > sourceStatusRefreshInterval {
-		u.Status(ctx)
+		u.updateSourceStatus(ctx)
```

Comment:
Still need to call Status here, because the metrics related variables are updated in Status.

Reply:
which metric?

Reply:
for example

```
			s.metricsProxies.Metrics.RemainingTimeGauge.Set(float64(remainingSeconds))
```

this line is contained in (*Syncer).printStatus <- (*Syncer).Status

Reply:
fixed here https://github.com/pingcap/tiflow/pull/7618/commits/c55d42df659a1026549397229f95600a2bc45937

bad idea to have a status getter to have such side effects.